connection_notifier.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "connection_notifier.hh"
+#include "db/query_context.hh"
+#include "cql3/constants.hh"
+#include "database.hh"
+#include "service/storage_proxy.hh"
+
+#include <stdexcept>
+
+namespace db::system_keyspace {
+const char *const clients_cf_name();
+}
+
+static sstring to_string(client_type ct) {
+    switch (ct) {
+        case client_type::cql: return "cql";
+        case client_type::thrift: return "thrift";
+        case client_type::alternator: return "alternator";
+        default: throw std::runtime_error("Invalid client_type");
+    }
+}
+
+future<> notify_new_client(client_data cd) {
+    const static sstring req
+            = format("INSERT INTO system.{} (address, port, client_type, shard_id, protocol_version, username) "
+                     "VALUES (?, ?, ?, ?, ?, ?);", db::system_keyspace::clients_cf_name());
+    
+    return db::execute_cql(req,
```

Comment:
Can we do that in a follow-up?
I would hesitate to prepare those statements without a benchmark. Not to mention that... this whole PR is temporary solution (until somebody understands/refactors virtual readers), so maybe it's not worth it..?

Reply:
This patch, temporary or not, will make connection establishment slower as each connection is logged. I *think* that because most clients will make an effort to only create a few connections, this overhead will be negligible, so your code is fine - but I don't really know.
I think it's important that if you feel that prepared statements could be better here but just don't want to do it now - that you should add a FIXME comment in the code, otherwise we'll never remember to go to it.

Reply:
Thanks, I added FIXME for now.
Anyway, I'm expected to write some test here later, so that will be a good opportunity for benchmark/prepare.